### PR TITLE
Jakartified modifications for 3.0.0-M2 release 

### PR DESCRIPTION
**Pull request by senivam**
Includes: 

 jakartified:
  jakarta.annotation
  jakarta.injection
  jakarta.validation
  org.glassfish.hk2

 Rebase from actual master (previous PR to the 3.x branch)

### CODE REVIEW

**Comment by senivam** [comment]
The latest commit adds jakartified dependencies for:
- weld, 
- jakarta.el, 
- org.glassfis.el, 
- jakarta.inject, 
- jakarta.validation, 
- org.hibernate.validator, 
- jakarta.enterprise.cdi, 
- jakarta.platform:jakatraee-api

File: bundles/jaxrs-ri/pom.xml [hunk 1]
```diff
@@ -322,7 +322,7 @@
                                         <exclude>META-INF/LICENSE.md</exclude>
                                         <exclude>javax/annotation/**</exclude>
                                         <exclude>javax/decorator/**</exclude>
-                                        <exclude>javax/el/**</exclude>
+                                        <exclude>jakarta.el/**</exclude>
```
Comment (jansupol) [inline bundles/jaxrs-ri/pom.xml:325]: `jakarta/el`, not `jakarta.el`

File: examples/cdi-webapp/pom.xml [hunk 2]
```diff
@@ -182,6 +199,18 @@
...
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <forkMode>always</forkMode>
+                    <enableAssertions>false</enableAssertions>
+                    <!-- TODO remove after jakartification -->
+                    <excludes>
+                        <exclude>org/glassfish/jersey/examples/cdi/resources/EchoParamBeanTest.java</exclude>
```
Comment (jansupol) [inline examples/cdi-webapp/pom.xml:210]: Not sure I follow. You have excluded all tests in `jakartification_exclude_tests` profile and here you exclude one more test extra....But ok.

File: examples/helloworld-cdi2-se/pom.xml [hunk 3]
```diff
@@ -81,6 +80,23 @@
                 </plugins>
             </build>
         </profile>
+        <!--<profile>
```
Comment (jansupol) [inline examples/helloworld-cdi2-se/pom.xml:83]: Do we need this when it's commented out?

File: inject/cdi2-se/pom.xml [hunk 4]
```diff
@@ -109,4 +114,23 @@
             </plugin>
         </plugins>
     </build>
+    <!--<profiles>
```
Comment (jansupol) [inline inject/cdi2-se/pom.xml:117]: Same here, do we need these comments?

File: pom.xml [hunk 5]
```diff
@@ -2129,25 +2118,26 @@
         <spring4.version>4.3.20.RELEASE</spring4.version>
         <spring5.version>5.1.5.RELEASE</spring5.version>
         <surefire.version>3.0.0-M3</surefire.version>
-        <validation.impl.version>6.0.17.Final</validation.impl.version>
+        <validation.impl.version>7.0.0.Alpha1</validation.impl.version>
         <weld.version>2.2.14.Final</weld.version> <!-- 2.4.1 doesn't work - bv tests -->
-        <weld3.version>3.0.0.Final</weld3.version>
+        <weld3.version>4.0.0.Alpha1</weld3.version>
```
Comment (jansupol) [inline pom.xml:2123]: I am afraid we need a CQ for this :(

File: tests/integration/cdi-integration/cdi-manually-bound/pom.xml [hunk 6]
```diff
@@ -80,6 +80,24 @@
         </dependency>
     </dependencies>
 
-
+    <!--<profiles>
```
Comment (jansupol) [inline tests/integration/cdi-integration/cdi-manually-bound/pom.xml:83]: Again comments?

File: examples/cdi-webapp/pom.xml (hunk 2 above)
Comment (senivam) [reply examples/cdi-webapp/pom.xml:210]: I would keep this the one removal refers to jakartification w/o Weld jakartified, and all tests are skipped due to jakartified weld

File: bundles/jaxrs-ri/pom.xml (hunk 1 above)
Comment (senivam) [reply bundles/jaxrs-ri/pom.xml]: fixed

File: tests/integration/cdi-integration/cdi-manually-bound/pom.xml (hunk 6 above)
Comment (senivam) [reply tests/integration/cdi-integration/cdi-manually-bound/pom.xml]: fixed

File: examples/helloworld-cdi2-se/pom.xml (hunk 3 above)
Comment (senivam) [reply examples/helloworld-cdi2-se/pom.xml]: fixed

**Comment by senivam** [comment]
CQs related to jakartification:
org.jboss.weld.se:weld-se-core:4.0.0.Alpha1 - [22129](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22129)
org.hibernate.validator:hibernate-validator:7.0.0.Alpha1 - [22131](http://dev.eclipse.org/ipzilla/show_bug.cgi?id=22131)

File: pom.xml (hunk 5 above)
Comment (senivam) [reply pom.xml:2123]: CQs are filled - references are in main comments

**Comment by smillidge** [comment]
GlassFish needs this change to start up. Any chance you can push a new milestone.

**Comment by jansupol** [comment]
@smillidge Sorry, we cannot go public until the CQs for Weld and BV are approved.

**Comment by arjantijms** [comment]
@jansupol 
> Sorry, we cannot go public until the CQs for Weld and BV are approved.

That's not necessary for just staging, right?

**Comment by smillidge** [comment]
Yeah GlassFish project only needs staging not maven central.

**Comment by senivam** [comment]
@smillidge , @arjantijms Jersey 3.0.0-M2 is staged. For now it is not possible to send it to central for 2 reasons: the 1st - mentioned CQs, and the 2nd - it has staged dependencies (which are not in central yet).

**Comment by smillidge** [comment]
Thanks from the GlassFish team. At the moment we don't need central as we are building from staging.

**Comment by senivam** [comment]
CQs are approved

**Pull request merged**